### PR TITLE
[11.0][ADD] hr_employee_medical_examination

### PR DESCRIPTION
**Pull request by jarroyomorales**
Hr Employee Medical Exam
========================

Adds information about employee's medical exams

Usage
=====

This module adds a wizard to generate medical exams for employees and an
action to see them.

The idea is to generate all the exams using the wizard and then manage them
individually even though the usage of the wizard is not mandatory.

For generating the medical exams:

#. Go to the menu *Employees > Medical Exams > Generate Medical Exams*.
#. Select a name and the employees and generate it.

To see all the medical exams:

#. Go to the menu *Employees > Medical Exams > View Medical Exams*.

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
@jarroyomorales I suggest 'hr_employee_medical_examination' instead of 'exam'. https://en.wikipedia.org/wiki/Physical_examination

I think that 'exam' is a wrong translation.

As I understand the wizard does not propose a date for the planned examination, which you need to enter manually at a later stage. But feels to me that it would be convenient if the user has the option to enter the date in the wizard.

**Comment by jarroyomorales** [comment]
@jbeficent I agree on the name change, wrong spanish translation.
And regarding the date, do you mean one date for each employee or one date on the wizard that applies to all employees?

**Review by JordiBForgeFlow — APPROVED** [review]
:+1:  technical and functional review.

**Comment by JordiBForgeFlow** [comment]
cc @hveficent can you review this one?

File: hr_employee_medical_examination/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Hr Employee Medical Examination',
+    'summary': """
+        Adds information about employee's medical examinations""",
+    'version': '11.0.1.0.0',
+    'license': 'AGPL-3',
+    'author': 'Creu Blanca,Odoo Community Association (OCA)',
+    'website': 'www.creublanca.es',
```
Comment (etobella) [inline hr_employee_medical_examination/__manifest__.py]: website, please

File: hr_employee_medical_examination/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    'depends': [
+        'hr'
```
Comment (etobella) [inline hr_employee_medical_examination/__manifest__.py]: missing comma

File: hr_employee_medical_examination/models/hr_employee_medical_examination.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class HrEmployeeMedicalExamination(models.Model):
+
+    _name = 'hr.employee.medical.examination'
+    _description = 'Hr Employee Medical Examination'
+    _inherit = ['mail.thread']
```
Comment (etobella) [inline hr_employee_medical_examination/models/hr_employee_medical_examination.py]: Why not activity? Maybe it is interesting to make a relation to an activity / reminder

File: hr_employee_medical_examination/wizards/wizard_generate_medical_examination.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                self.env['hr.employee.medical.examination'].create({
+                    'name': '%s on %s' % (form.name, employee.name),
+                    'employee_id': employee.id,
+                })
+        action = self.env.ref(
+            'hr_employee_medical_examination.hr_employee'
+            '_medical_examination_act_window',
+            False
+        )
+        result = action.read()[0]
```
Comment (etobella) [inline hr_employee_medical_examination/wizards/wizard_generate_medical_examination.py:69]: Maybe you should filter for the generated examinations.

File: hr_employee_medical_examination/wizards/wizard_generate_medical_examination.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            'res_id': self.id,
+            'context': self._context,
+        }
+        return action
+
+    @api.multi
+    def create_medical_examinations(self):
+        for form in self:
+            for employee in form.employee_ids:
+                self.env['hr.employee.medical.examination'].create({
```
Comment (etobella) [inline hr_employee_medical_examination/wizards/wizard_generate_medical_examination.py]: Put the values for creation on a hook in order to allow modifications please

**Review by etobella — CHANGES_REQUESTED** [review]

**Comment by jarroyomorales** [comment]
@etobella Comments attended!

**Review by etobella — APPROVED** [review]
LGTM :+1:

**Comment by etobella** [comment]
> As I understand the wizard does not propose a date for the planned examination, which you need to enter manually at a later stage. But feels to me that it would be convenient if the user has the option to enter the date in the wizard.

 @jbeficent When you program your employees medical examinations they don't do the examination all on the same day, because it might be hard to stop your company for 4 hours. Also, people could refuse to do the examination, So I think it is more an employee configuration.

**Review by HviorForgeFlow — COMMENTED** [review]
Functional review. LGTM. I miss some notes field, it could be done with chatter instead, but I think a note could be fine.

Also a pending filter.

**Comment by jarroyomorales** [comment]
@hveficent  Added a filter by name, employee and result.
Regarding the notes, I think it is better to use the chatter so people interested on an examination can subscibre to it and receive notifications when someone write on its chatter.

File: hr_employee_medical_examination/views/hr_employee_medical_examination_views.xml [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    <record model="ir.ui.view" id="hr_employee_medical_examination_search_view">
+        <field name="name">hr.employee.medical.examination.search (in
+            hr_employee_medical_examination)
+        </field>
+        <field name="model">hr.employee.medical.examination</field>
+        <field name="arch" type="xml">
+            <search>
+                <field name="name"/>
+                <field name="employee_id"/>
+                <group>
```
Comment (HviorForgeFlow) [inline hr_employee_medical_examination/views/hr_employee_medical_examination_views.xml:52]: I was asking by pending filter here:
```suggestion
                <group>
                    <filter string="Pending" name="pending" domain="[('state','=','pending')]"/>
```
It's up to your criteria, non blocking

**Review by HviorForgeFlow — APPROVED** [review]

File: hr_employee_medical_examination/views/hr_employee_medical_examination_views.xml (hunk 6 above)
Comment (jarroyomorales) [reply hr_employee_medical_examination/views/hr_employee_medical_examination_views.xml:52]: Oh, I missunderstood, just added the pending filter.

**Comment by jarroyomorales** [comment]
@jbeficent @pedrobaeza @alexey-pelykh Can this be merged? Thank you!

**Review by alexey-pelykh — APPROVED** [review]

**Comment by alexey-pelykh** [comment]
Code looks good to me

**Pull request merged**